app/services/data_migrations/create_or_update_provider_user_notification_preferences.rb

```diff
@@ -0,0 +1,19 @@
+module DataMigrations
+  class CreateOrUpdateProviderUserNotificationPreferences
+    TIMESTAMP = 20210315142623
+    MANUAL_RUN = false
+
+    def change
+      ProviderUser.find_each do |provider_user|
+        preferences = ProviderUserNotificationPreferences.find_or_initialize_by(provider_user: provider_user)
+        preferences.application_received = provider_user.send_notifications
```

Comment:
I was going to suggest using `update_all_preferences` on `ProviderUserNotificationPreferences `, but noticed that it is looping over each notification and performing a query for each, maybe we can improve its performance by assigning all the notifications and update them in one transaction instead, and then we can reuse here as well?